paimon-core/src/main/java/org/apache/paimon/table/AbstractFileStoreTable.java

```diff
@@ -82,6 +86,7 @@ public abstract class AbstractFileStoreTable implements FileStoreTable {
     protected final TableSchema tableSchema;
     protected final Lock.Factory lockFactory;
     @Nullable protected final MetastoreClient.Factory metastoreClientFactory;
+    private Set<Object> metaPaths = new HashSet<>(Arrays.asList("manifest", "schema"));
```

Comment:
I think we need to think about what truncate needs to keep, users should expect to get a brand new table when using truncate

Reply:
I think the truncate table should be a brand new table that does not contain historical data. For example, hudi directly deletes data files, which has much higher performance than [#1459](https://github.com/apache/incubator-paimon/pull/1459)  per data traversal